armi/reactor/components/component.py

```diff
@@ -1395,7 +1401,8 @@ def getPinIndices(self) -> Optional[np.ndarray[tuple[int], np.uint16]]:
         for sibling in withPinIndices:
             if sibling.spatialLocator == self.spatialLocator:
                 return sibling.p.pinIndices
-        return None
+        msg = f"{self} on {self.parent} has no pin indices."
+        raise ValueError(msg)
```

Comment:
I'm pretty soft on this change. I think returning a numpy array of length zero or None will be just as informative and easy to handle downstream.

But if it's important to you, you can make folks downstream handle the error.